crates/bevy_ecs/src/relationship/related_methods.rs

```diff
@@ -34,6 +34,39 @@ impl<'w> EntityWorldMut<'w> {
         self
     }
 
+    /// Replaces all the related entities with the given set of new related entities.
+    pub fn replace_related<R: Relationship>(&mut self, related: &[Entity]) -> &mut Self {
+        if related.is_empty() {
```

Comment:
This seems incorrect: replacing the children with an empty list should remove all existing children.